### PR TITLE
Drop HTTP-SOTA support

### PR DESCRIPTION
**Pull request by OYTIS**
A better observable codebase should make refactoring easier.

### CODE REVIEW

**Comment by patriotyk** [comment]
Not all code has been removed. For example authenticate method of http. Also we have some specific events and commands. Also download method of http class is also specific. And config options.

**Comment by OYTIS** [comment]
@patriotyk
Thank you for the review, I've tried to track all the unused events and commands and delete them. I've also dropped support for socketgateway after a discussion with @cajun-rat . Please take a look if I've forgotten something or maybe dropped something which is really important to us.

**Comment by OYTIS** [comment]
@cajun-rat @patriotyk Returned socketgateway until we can discuss its deletion with @christianhuggerATS

**Comment by OYTIS** [comment]
@patrickvacek 
Can't pass check-format. I've tried clang-format v3.9.1-2 and v3.8.1-1.

**Comment by cajun-rat** [comment]
Tried reformatting using the version on my machine.  I had assumed that its behaviour would be more stable...

**Comment by patrickvacek** [comment]
Rather disappointing that clang-format would yield such differences between seemingly similar versions. Could this be a configuration issue? I would hope we could rely on some amount of consistency.

File: src/eventsinterpreter.cc [hunk 1]
```diff
@@ -29,17 +29,6 @@ void EventsInterpreter::run() {
     gateway_manager.processEvents(event);
 
     if (event->variant == "DownloadComplete") {
```
Comment (patriotyk) [inline src/eventsinterpreter.cc]: I think we don't need such empty blocks

**Comment by OYTIS** [comment]
@patriotyk Done, thanks.

**Pull request merged**